sktime/datatypes/_series/_check.py

```diff
@@ -235,3 +236,71 @@ def _index_equally_spaced(index):
     all_equal = np.all(diffs == diffs[0])
 
     return all_equal
+
+
+if _check_soft_dependencies("xarray", severity="none"):
+    import xarray as xr
+
+    def check_xrdataarray_series(obj, return_metadata=False, var_name="obj"):
+        metadata = {}
+
+        def ret(valid, msg, metadata, return_metadata):
+            if return_metadata:
+                return valid, msg, metadata
+            return valid
+
+        if not isinstance(obj, xr.DataArray):
+            msg = f"{var_name} must be a xarray.DataArray, found {type(obj)}"
+            return ret(False, msg, None, return_metadata)
+
+        # we now know obj is a xr.DataArray
+        if len(obj.dims) > 2:  # Without multi indexing only two dimensions are possible
+            msg = f"{var_name} must have two or less dimension, found {type(obj.dims)}"
+            return ret(False, msg, None, return_metadata)
+
+        # The first dimension is the index of the time series in sktimelen
+        index = obj.indexes[obj.dims[0]]
```

Comment:
Where should I document the required format of xr.DataArray? Currently, the xr.DataArray must have two dimensions. The first dimension is the index, and the second dimension describes the columns.

Reply:
currently I've collected specifications in `examples/AA_datatypes`, although this is a bit ad-hoc, and long-term we may at some point like to split the "exhaustive list" (specification reference) from a "tutorial notebook". For now, I'd add it there, or perhaps a second notebook which has the less commonly used mtypes (too many mtypes would dilute the tutorial style experience)?